packages/devextreme/testing/tests/DevExpress.ui.widgets/menu.tests.js

```diff
@@ -674,6 +674,150 @@ QUnit.module('Rendering Scrollablew', {
+            .press('left')
+            .press('right')
+            .press('down');
+
+        assert.roughEqual($scrollableContent.position().top, 0, 1, 'scroll position reset');
+    });
+
+    QUnit.test('Scroll position should be set to 0 after reopen (nested submenu)', function(assert) {
+        if(!isDeviceDesktop(assert)) {
+            return;
+        }
+
+        const menu = createMenuInWindow({
+            items: [
+                {
+                    text: 'Item 1',
+                    items: [
+                        {
+                            text: 'Item 11',
+                            items: (new Array(99)).fill(null).map((_, idx) => ({ text: `item ${idx}` })),
+                        },
+                        {
+                            text: 'Item 22',
+                        },
+                    ],
+                },
+            ],
+            showFirstSubmenuMode: 'onClick',
+            showSubmenuMode: { name: 'onHover', delay: 0 },
+        });
+        const itemsContainer = menu.instance.itemsContainer();
+
+        keyboardMock(itemsContainer)
+            .press('down')
+            .press('right');
+
+        const $nestedSubmenu = $(`.${DX_SUBMENU_CLASS}`).eq(1);
+        const $scrollableContainer = $nestedSubmenu.find(`.${DX_SCROLLABLE_CONTAINER_CLASS}`);
+        const $scrollableContent = $nestedSubmenu.find(`.${DX_SCROLLABLE_CONTENT_CLASS}`);
+
+        keyboardMock(itemsContainer)
+            .press('up');
+
+        assert.roughEqual($scrollableContent.position().top,
+            $scrollableContainer.height() - $scrollableContent.height() + BORDER_WIDTH, 1, 'scrolled to bottom');
+
+        keyboardMock(itemsContainer)
+            .press('left')
+            .press('right');
+
+        assert.roughEqual($scrollableContent.position().top, 0, 1, 'scroll position is reset (KBN)');
+
+        keyboardMock(itemsContainer)
+            .press('up');
+
+        assert.roughEqual($scrollableContent.position().top,
+            $scrollableContainer.height() - $scrollableContent.height() + BORDER_WIDTH, 1, 'scrolled to bottom');
+
+        const $rootMenuItem = $(menu.element).find(`.${DX_MENU_ITEM_CLASS}`).eq(0);
+        const submenu = getSubMenuInstance($rootMenuItem);
+        const $firstMenuItem = $(submenu._overlay.content()).find(`.${DX_MENU_ITEM_CLASS}`).first();
+        const $lastMenuItem = $(submenu._overlay.content()).find(`.${DX_MENU_ITEM_CLASS}`).last();
+
+        $(submenu.itemsContainer()).trigger($.Event('dxhoverstart', { target: $lastMenuItem.get(0) }));
+        $lastMenuItem.eq(0).trigger('dxpointermove');
+        this.clock.tick(0);
+        $(submenu.itemsContainer()).trigger($.Event('dxhoverstart', { target: $firstMenuItem.get(0) }));
+        $firstMenuItem.eq(0).trigger('dxpointermove');
+        this.clock.tick(0);
+
+        assert.roughEqual($scrollableContent.position().top, 0, 1, 'scroll position is reset (hover)');
+    });
```

Comment:
please split the tests, they should be atomic. Now it checks 2 different scenarios, that's why it's so huge